src/components/CartItems.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux'
+import { ListGroup } from 'react-bootstrap';
+import { AppDispatch } from '../actions';
+import { AppState } from "../store";
+import { CartViewModifier } from '../types/cart';
+import { removeFromCart } from '../actions/cart';
+
+type Props = {
+  cartView?: CartViewModifier
+}
+
+const CartItems: React.FC<Props> = ({ cartView = CartViewModifier.NONE }) => {
+  const products = useSelector((state: AppState) => state.cartState.products);
+  const dispatch = useDispatch<AppDispatch>();
+  const cartTotal = products.reduce((total, prod) => total + prod.price, 0);
+  return (
+    <ListGroup>
+      { products.map((product, i) => (
+        <ListGroup.Item key={i}>
+          {product.title}
+          <span className='cart-price'>${product.price} <span className='text-muted'>{product.unit}</span>
+            {CartViewModifier.DELETE === (cartView & CartViewModifier.DELETE) &&
```

Comment:
@reviewer I'd love to know your thoughts on this approach: I'm using a bitwise enum as a feature switch. In this example, it's not really necessary (I could have used a boolean `showDelete` prop), but if a component had several permutations of additional controls, etc., that were dependent on some context (like RBAC state), I can see this being a concise way to control the view of a component.

Reply:
It's definitely a cool approach! Though, if it's not being used, might be less cognitive overhead to use a plain enum and match exactly based on that state?

Reply:
I'll do that. I wanted to get this idea in front of you and get your thoughts. I've seen a few talks/articles about prop explosion for component view permutations and thought this was an interesting solution.

Grouping bitwise enum values for RBAC control of views seems like a great use case.

Reply:
Using vanilla enum.